File: tutorials/composites/src/main/java/org/qi4j/tutorials/composites/tutorial6/HelloWorldComposite.java
package org.qi4j.tutorials.composites.tutorial6;

import org.qi4j.api.composite.TransientComposite;

/**
 * This Composite interface declares transitively
 * all the Fragments of the HelloWorld composite.
 * <p/>
 * What Mixins to use and what Concerns should
 * apply to the methods can be found by exploring
 * the interfaces extended by this Composite interface,
 * and by looking at the declared @Mixins annotations.
 */
public interface HelloWorldComposite
    extends HelloWorld, TransientComposite
{
}

File: libraries/lang-scala/src/test/scala/org/qi4j/sample/scala/HelloWorldComposite.scala
package org.qi4j.sample.scala

import org.qi4j.api.composite.TransientComposite
import org.qi4j.api.concern.Concerns

@Concerns(Array(classOf[ HelloThereConcern ]))
trait HelloWorldComposite
  extends TransientComposite with HelloWorldMixin with HelloWorldMixin2
